lib/elixir/lib/calendar.ex

```diff
@@ -314,6 +321,23 @@ defmodule Date do
     end
   end
 
+  @doc """
+  Computes the day of the week of a given `Date` struct.
+
+  Returns the day of the week as 1: Monday, 2: Tuesday, and so on.
+
+  ## Examples
+
+      iex> Date.day_of_week(~D[2016-10-31])
+      1
+      iex> Date.day_of_week(~D[2016-11-01])
+      2
+  """
+  @spec day_of_week(Date.t) :: integer
```

Comment:
yes. sorry i missed the whole `calendar` variable thing.
So what about the ISO implementation?